src/linker/pipeline.py

```diff
@@ -18,14 +19,53 @@ def __init__(self, config: Config):
         self._validation_errors = {}
         self._validate()
 
-    def run(self, runner: Callable, results_dir: Path) -> None:
-        for implementation in self.implementations:
+    def run(
+        self,
+        runner: Callable,
+        results_dir: Path,
+        log_dir: Path,
+        session: Optional[types.ModuleType("drmaa.Session")],
+    ) -> None:
+        number_of_steps = len(self.implementations)
+        for idx, implementation in enumerate(self.implementations):
+            output_dir = (
+                results_dir
+                if idx == (number_of_steps - 1)
+                else results_dir / "intermediate" / implementation.name
```

Comment:
This is a nit, but I think the intermediate results directories should use the _step_ name, either alone or in addition to the implementation name. I don't want us to force the implementation names to be unique.